What is the way to declare an endpoint that returns a flask.Response object?

I'm trying to create an endpoint that returns a flask.Response (302).
Code:
```
from flask import redirect
def dummy():
    return redirect('/')
```
But I get the following error:
```
<Response 209 bytes [302 FOUND]> is not JSON serializable
```
Additionally if I try to avoid the JSON conversion setting the output for that endpoint as text/plain using the following swagger definition:
```
  /dummy:
    get:
      produces: "text/plain"
      operationId: controllers.dummy.dummy
      responses:
        302:
          description: OK
```
I get the following stack trace when I launch the server:
```
test_1 | ERROR:connexion.api:Failed to add operation for GET /v1/dummy
test_1 | Traceback (most recent call last):
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/api.py", line 136, in add_paths
test_1 |     self.add_operation(method, path, endpoint)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/api.py", line 122, in add_operation
test_1 |     self.blueprint.add_url_rule(flask_path, operation.endpoint_name, operation.function, methods=[method])
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 234, in function
test_1 |     produces_decorator = self.__content_type_decorator
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 271, in __content_type_decorator
test_1 |     mimetype = self.get_mimetype()
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 165, in get_mimetype
test_1 |     if produces_json(self.produces):  # endpoint will return json
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 121, in produces_json
test_1 |     return all(is_json_mimetype(mimetype) for mimetype in produces)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 121, in <genexpr>
test_1 |     return all(is_json_mimetype(mimetype) for mimetype in produces)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 95, in is_json_mimetype
test_1 |     maintype, subtype = mimetype.split('/')  # type: str, str
test_1 | ValueError: need more than 1 value to unpack
test_1 | ERROR:connexion.api:Failed to add operation for GET /v1/dummy
test_1 | Traceback (most recent call last):
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/api.py", line 136, in add_paths
test_1 |     self.add_operation(method, path, endpoint)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/api.py", line 122, in add_operation
test_1 |     self.blueprint.add_url_rule(flask_path, operation.endpoint_name, operation.function, methods=[method])
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 234, in function
test_1 |     produces_decorator = self.__content_type_decorator
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 271, in __content_type_decorator
test_1 |     mimetype = self.get_mimetype()
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/operation.py", line 165, in get_mimetype
test_1 |     if produces_json(self.produces):  # endpoint will return json
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 121, in produces_json
test_1 |     return all(is_json_mimetype(mimetype) for mimetype in produces)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 121, in <genexpr>
test_1 |     return all(is_json_mimetype(mimetype) for mimetype in produces)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/utils.py", line 95, in is_json_mimetype
test_1 |     maintype, subtype = mimetype.split('/')  # type: str, str
test_1 | ValueError: need more than 1 value to unpack
```

What is the right way to return a flask.Response object using connexion?

Check https://github.com/jvgutierrez/connexion/commit/3697a5d72c40795fe5600491a609f8421d11322d
```
tests/test_app.py::test_redirect_endpoint PASSED
tests/test_app.py::test_redirect_response_endpoint FAILED
```
```
__________________________________________________________________________ test_redirect_response_endpoint __________________________________________________________________________

app = <connexion.app.App instance at 0x7f531002d4d0>

    def test_redirect_response_endpoint(app):
        app_client = app.app.test_client()
        resp = app_client.get('/v1.0/test-redirect-response-endpoint')
>       assert resp.status_code == 302
E       assert 500 == 302
E        +  where 500 = <Response streamed [500 INTERNAL SERVER ERROR]>.status_code
```
```
------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------
DEBUG:connexion.decorators.produces:Jsonifing http://localhost/v1.0/test-redirect-response-endpoint
DEBUG:connexion.decorators.parameter:Function Arguments: []
DEBUG:connexion.decorators.decorator:Getting data and status code
DEBUG:connexion.decorators.decorator:Got data and status code (200)
DEBUG:connexion.decorators.decorator:Getting data and status code
DEBUG:connexion.decorators.decorator:Got data and status code (200)
ERROR:test_app:Exception on /v1.0/test-redirect-response-endpoint [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/app/connexion/decorators/produces.py", line 126, in wrapper
    data = json.dumps(data, indent=2, cls=JSONEncoder)
  File "/usr/local/lib/python2.7/json/__init__.py", line 251, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/local/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/local/lib/python2.7/json/encoder.py", line 442, in _iterencode
    o = _default(o)
  File "/app/connexion/decorators/produces.py", line 43, in default
    return json.JSONEncoder.default(self, o)
  File "/usr/local/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <Response 251 bytes [302 FOUND]> is not JSON serializable
```

--- a/connexion/decorators/decorator.py
+++ b/connexion/decorators/decorator.py
@@ -12,6 +12,7 @@ Unless required by applicable law or agreed to in writing, software distributed 
 """
 import logging
 import flask
+from ..utils import is_flask_response
 
 logger = logging.getLogger('connexion.decorators.decorator')
 
@@ -32,7 +33,7 @@ class BaseDecorator:
         url = flask.request.url
         logger.debug('Getting data and status code', extra={'data': data, 'data_type': type(data), 'url': url})
         status_code, headers = 200, {}
-        if isinstance(data, flask.Response):
+        if is_flask_response(data):
             data = data
             status_code = data.status_code
             headers = data.headers

--- a/connexion/decorators/produces.py
+++ b/connexion/decorators/produces.py
@@ -18,6 +18,7 @@ import functools
 import json
 import logging
 from .decorator import BaseDecorator
+from ..utils import is_flask_response
 
 logger = logging.getLogger('connexion.decorators.produces')
 
@@ -83,7 +84,7 @@ class Produces(BaseSerializer):
             url = flask.request.url
             data, status_code, headers = self.get_full_response(function(*args, **kwargs))
             logger.debug('Returning %s', url, extra={'url': url, 'mimetype': self.mimetype})
-            if isinstance(data, flask.Response):  # if the function returns a Response object don't change it
+            if is_flask_response(data):
                 logger.debug('Endpoint returned a Flask Response', extra={'url': url, 'mimetype': data.mimetype})
                 return data
 
@@ -114,7 +115,7 @@ class Jsonifier(BaseSerializer):
             url = flask.request.url
             logger.debug('Jsonifing %s', url, extra={'url': url, 'mimetype': self.mimetype})
             data, status_code, headers = self.get_full_response(function(*args, **kwargs))
-            if isinstance(data, flask.Response):  # if the function returns a Response object don't change it
+            if is_flask_response(data):
                 logger.debug('Endpoint returned a Flask Response', extra={'url': url, 'mimetype': data.mimetype})
                 return data
             elif data is NoContent:

--- a/connexion/utils.py
+++ b/connexion/utils.py
@@ -14,6 +14,8 @@ Unless required by applicable law or agreed to in writing, software distributed 
 import functools
 import importlib
 import re
+import flask
+import werkzeug.wrappers
 
 PATH_PARAMETER = re.compile(r'\{([^}]*)\}')
 
@@ -57,6 +59,21 @@ def flaskify_path(swagger_path, types={}):
     """
     convert_match = functools.partial(convert_path_parameter, types=types)
     return PATH_PARAMETER.sub(convert_match, swagger_path)
+
+
+def is_flask_response(obj):
+    """
+    Verifies if obj is a default Flask response instance.
+
+    :type obj: object
+    :rtype bool
+
+    >>> is_flask_response(redirect('http://example.com/'))
+    True
+    >>> is_flask_response(flask.Response())
+    True
+    """
+    return isinstance(obj, flask.Response) or isinstance(obj, werkzeug.wrappers.Response)
 
 
 def deep_getattr(obj, attr):

--- a/tests/fakeapi/api.yaml
+++ b/tests/fakeapi/api.yaml
@@ -656,6 +656,22 @@ paths:
           in: query
           required: true
 
+  /test-redirect-endpoint:
+    get:
+      summary: Tests handlers returning flask.Response objects
+      operationId: fakeapi.hello.test_redirect_endpoint
+      responses:
+        302:
+          description: 302 Found
+
+  /test-redirect-response-endpoint:
+    get:
+      summary: Tests handlers returning flask.Response objects
+      operationId: fakeapi.hello.test_redirect_response_endpoint
+      responses:
+        302:
+          description: 302 Found
+
 definitions:
   new_stack:
     type: object

--- a/tests/fakeapi/hello.py
+++ b/tests/fakeapi/hello.py
@@ -2,6 +2,7 @@
 
 from connexion import problem, request
 from connexion import NoContent
+from flask import redirect
 
 
 class DummyClass:
@@ -234,3 +235,12 @@ def test_bool_default_param(thruthiness):
 
 def test_required_param(simple):
     return simple
+
+
+def test_redirect_endpoint():
+    headers = {'Location': 'http://www.google.com/'}
+    return '', 302, headers
+
+
+def test_redirect_response_endpoint():
+    return redirect('http://www.google.com/')

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -613,3 +613,15 @@ def test_required_param_miss_config(app):
 
     resp = app_client.get('/v1.0/test-required-param')
     assert resp.status_code == 400
+
+
+def test_redirect_endpoint(app):
+    app_client = app.app.test_client()
+    resp = app_client.get('/v1.0/test-redirect-endpoint')
+    assert resp.status_code == 302
+
+
+def test_redirect_response_endpoint(app):
+    app_client = app.app.test_client()
+    resp = app_client.get('/v1.0/test-redirect-response-endpoint')
+    assert resp.status_code == 302

